website/content/docs/commands/setup/vault.mdx

```diff
@@ -0,0 +1,153 @@
+</Warning>
+
+## Usage
+
+```plaintext
+nomad setup vault [options]
+```
+
+## General Options
+
+@include 'general_options_no_namespace.mdx'
+
+## Setup Vault Options
+
+- `-jwks-url`: URL of Nomad's JWKS endpoint contacted by Consul to verify JWT
+  signatures. Defaults to `http://localhost:4646/.well-known/jwks.json`.
+
+- `-cleanup`: Removes all configuration components this command created from the
+  Consul cluster.
+
+- `-y`: Automatically answers "yes" to all the questions, making the setup
+  non-interactive. Defaults to "false".
+
+## Examples
+
+Below is an example of an interactive session with default options, interrupted
+by answering "no" to one of the questions, demonstrating the helper's "cleanup"
+capabilities as well.
+
+```shell-session
+$ nomad setup vault
+
+This command will walk you through configuring all the components required for
+Nomad workloads to authenticate themselves against Vault ACL using their
+respective workload identities.
+
+First we need to connect to Vault.
+
+[?] Is "http://127.0.0.1:8200" the correct address of your Vault cluster? [Y/n]
+
+Since you're running Vault Enterprise, we will additionally create
+a namespace "nomad-workloads" and create all configuration within that namespace.
+
+[?] Create the namespace "nomad-workloads" in your Vault cluster? [Y/n]
+[✔] Created namespace "nomad-workloads".
+
+We will now enable the JWT credential backend and create a JWT auth method that
+Nomad workloads will use.
+
+This is the method configuration:
+
+{
+    "default_role": "nomad-workloads",
+    "jwks_url": "http://localhost:4646/.well-known/jwks.json",
+    "jwt_supported_algs": [
+        "EdDSA",
+        "RS256"
+    ]
+}
+[?] Create JWT auth method in your Vault cluster? [Y/n]
+[✔] Created JWT auth method "jwt-nomad".
+
+We need to create a role that Nomad workloads will assume while authenticating,
+and a policy associated with that role.
+
+
+These are the rules for the policy "nomad-workloads" that we will create. It uses a templated
+policy to allow Nomad tasks to access secrets in the path
+"secrets/data/<job namespace>/<job name>":
+
+path "secret/data/{{identity.entity.aliases.auth_jwt_1b8dcc32.metadata.nomad_namespace}}/{{identity.entity.aliases.auth_jwt_1b8dcc32.metadata.nomad_job_id}}/*" {
+  capabilities = ["read"]
+}
+
+path "secret/data/{{identity.entity.aliases.auth_jwt_1b8dcc32.metadata.nomad_namespace}}/{{identity.entity.aliases.auth_jwt_1b8dcc32.metadata.nomad_job_id}}" {
+  capabilities = ["read"]
+}
+
+path "secret/metadata/{{identity.entity.aliases.auth_jwt_1b8dcc32.metadata.nomad_namespace}}/*" {
+  capabilities = ["list"]
+}
+
+path "secret/metadata/*" {
+  capabilities = ["list"]
+}
+
+[?] Create the above policy in your Vault cluster? [Y/n]
+[✔] Created policy "nomad-workloads".
+
+We will now create an ACL role called "nomad-workloads" associated with the policy above.
+
+{
+    "bound_audiences": "vault.io",
+    "claim_mappings": {
+        "nomad_group": "nomad_group",
```

Comment:
Ops, we actually need to remove this as I realized that we don't actually have this claim for any workload identity. I thought it was just for services 😬 

No harm in having it though, it just never exists in JWTs.